### PR TITLE
Add fhir migrations and new SearchParameter

### PR DESCRIPTION
**Pull request by pbugni**
https://www.pivotaltracker.com/n/projects/2493717/stories/188376912 requires a new SearchParameter.

Bringing [fhir-migrations](https://github.com/uwcirg/fhir-migrations) into cosri dashboard for easier maintenance of FHIR resource changes.

Includes first migration script `next_appt_search_parameter.py` to satisfy story need.

TODO: consider extending Dockerfile to run `flask upgrade` on every deploy.
NB: depends on https://github.com/uwcirg/cosri-patientsearch/pull/230

### CODE REVIEW

File: base/freestanding/femr/docker-compose.yaml [hunk 1]
```diff
@@ -30,6 +33,8 @@ services:
     networks:
       - ingress
       - internal
+    volumes:
+      - ./config/migrations:/opt/migrations
```
Comment (ivan-c) [inline base/freestanding/femr/docker-compose.yaml:37]: minor nitpick: I generally set a bind-mount as read-only, if I know the container doesn't need to write to the host. it's a minor improvement in security, but mostly for preventing unexpected changes eg if you delete files inside a container during testing


```suggestion
      - ./config/migrations:/opt/migrations:ro
```

File: base/freestanding/femr/config/migrations/next_appt_search_parameter.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        "code": "date-time-of-next-appointment",
+        "base": [ "Patient" ],
+        "type": "date",
+        "expression": "Patient.extension('http://www.uwmedicine.org/time_of_next_appointment').valueDateTime"
+    }
+    response = requests.put(f'{FHIR_SERVER_URL}SearchParameter/{SP_ID}', headers=HEADERS, data=json.dumps(sp_resource))
+    if response.status_code == 200 or response.status_code == 201:
+        logging.info('SearchParameter created successfully.')
+    else:
+        logging.error(f'Failed to create SearchParameter: {response.status_code} {response.text}')
```
Comment (ivan-c) [inline base/freestanding/femr/config/migrations/next_appt_search_parameter.py:40]: I might be missing some nuance in the HTTP codes returned by Hapi, but I always like this catch-all
```suggestion
    response.raise_for_status()
    logging.info('SearchParameter created successfully.')
```

**Review by ivan-c — APPROVED** [review]
Looks great, excited to roll one of these out!!

File: base/freestanding/femr/docker-compose.yaml (hunk 1 above)
Comment (pbugni) [reply base/freestanding/femr/docker-compose.yaml:37]: actually, it does need to write there, in the event of using fhir-migrations to generate a migration template script via `flask migrate <name>`

File: base/freestanding/femr/config/migrations/next_appt_search_parameter.py (hunk 2 above)
Comment (pbugni) [reply base/freestanding/femr/config/migrations/next_appt_search_parameter.py:40]: agreed!  i had used the examples but prefer this approach.  fhir-migrations will get better with use.

Comment (mcjustin) [reply base/freestanding/femr/config/migrations/next_appt_search_parameter.py:40]: Funny, I just ran into such a Hapi nuance at https://github.com/uwcirg/fhir-attrib-setter/blob/main/fhir-attrib-setter.patient.extension.py#L93 , Hapi might respond w/ a 200 code even though it had an error and didn't write anything. json for such a resonse: 
```
{"resourceType": "OperationOutcome", "issue": [{"severity": "error", "code": "processing", "diagnostics": "HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element=\"valueDateTime\"] Invalid attribute value \"2044-10-11 09:57:27\": Invalid date/time format: \"2044-10-11 09:57:27\": Expected character 'T' at index 10 but found  "}]}
```

**Pull request merged**